contracts/adapters/ibc/ibc-hooks/src/contract.rs

```diff
@@ -9,29 +9,46 @@ use cosmwasm_std::{
     entry_point, to_binary, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Env, MessageInfo,
     Reply, Response, SubMsg, SubMsgResult,
 };
+use cw2::set_contract_version;
 use ibc_proto::ibc::applications::transfer::v1::{MsgTransfer, MsgTransferResponse};
 use prost::Message;
 use serde_cw_value::Value;
 use skip::{
-    ibc::{AckID, ExecuteMsg, IbcInfo, IbcLifecycleComplete, InstantiateMsg, QueryMsg},
+    ibc::{AckID, ExecuteMsg, IbcInfo, IbcLifecycleComplete, InstantiateMsg, MigrateMsg, QueryMsg},
     proto_coin::ProtoCoin,
     sudo::{OsmosisSudoMsg as SudoMsg, SudoType},
 };
 
 const IBC_MSG_TRANSFER_TYPE_URL: &str = "/ibc.applications.transfer.v1.MsgTransfer";
 const REPLY_ID: u64 = 1;
 
+///////////////
+/// MIGRATE ///
+///////////////
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> ContractResult<Response> {
+    unimplemented!()
```

Comment:
Does this not block the upgrade by panicking? Don't you want to just return successfully here?

Reply:
Migrations work using the migration code of the new contract you are migrating to, not of the deployed contract. So when we instantiate all the contracts currently in the repo, they will all deploy without every calling their migrate functions. But then if we want to upgrade a contract in the future, we will do so by implementing that contracts migrate function, store that code on chain, and then submit the MigrateMsg

Reply:
tldr: we only implement the migrate function when we are doing a migration/upgrade, but can keep them unimplemented on normal deploys.